h2o-algos/src/main/java/hex/tree/DHistogram.java

```diff
@@ -344,15 +361,23 @@ else if (_histoType== HistogramType.QuantilesGlobal) {
               _histoType = HistogramType.UniformAdaptive;
             }
             else {
-              _hasQuantiles=true;
+              _absoluteSplitPts = true;
               _nbin = (char)_splitPts.length;
               if (LOG.isTraceEnabled()) LOG.trace("Refined splitPoints: " + Arrays.toString(_splitPts));
             }
           }
         }
       }
     }
-    else assert(_histoType== HistogramType.UniformAdaptive);
+    else if (_histoType == HistogramType.UniformRobust) {
+      if (_customSplitPoints != null) {
+        defineSplitPointsFromCustomSplitPoints(_customSplitPoints);
+      } else {
+        _histoType = HistogramType.UniformAdaptive;
```

Comment:
There should be no warnings in the DHistogram code - this is high perf code, it would pollute the logs and hurt performance. 

This is IMHO the same behavior as with QuantilesGlobal. Can you please suggest how to clarify the code? (Or should we add a comment?)

Reply:
Ahh, I see now what is the purpose. But comments are always welcome. :)

Reply:
Thank you, I added a comment explaining the behavior.